private/buf/buffetch/internal/reader.go

```diff
@@ -361,6 +363,9 @@ func (r *reader) getProtoFileBucket(
 	if !r.localEnabled {
 		return nil, NewReadLocalDisabledError()
 	}
+	if datawkt.Exists(protoFileRef.Path()) {
```

Comment:
This means that you will always prioritize the built-in well-known types over whatever is on disk. As a counterexample of how this could break down, assume I have an old, incompatible version of `google/protobuf/timestamp.proto` on disk at `./google/protobuf/timestamp.proto`, and I specify `buf beta convert google/protobuf/timestamp.proto`. With this code, I will always use the built-in version and not the one on disk, whereas the rest of our code prioritizes the one on disk over the builtins.